### PR TITLE
Fix: Do not start protocol adapter if not enough memory.

### PR DESCRIPTION
**Pull request by b-abel**
Throw a exception if the connection limit is auto-configured to 0. 
Currently the protocol adapter only prints a warning to the logs, but then completes the startup. So, possibly nobody would notice that it would not accept any connections.

### CODE REVIEW

**Comment by sophokles73** [comment]
I think it is reasonable to indicate this problem to the user more explicitly, as proposed in this PR.
I also wonder if the current strategy we follow, i.e. shutting down the service and exiting the JVM, if a problem occurs, is very helpful. Most container orchestration platforms will immediately try to restart the container which will most probably fail again and then over and over again. This makes it particularly hard for ops people to get to the cause of the problem because the log of the exited container might already be gone as well. Maybe it would therefore be better to start the container successfully, but never let it reach the *ready* state instead. WDYT? @ctron @dejanb

**Comment by sophokles73** [comment]
@b-abel can you please rebase your PR so that the Travis build has a chance of succeeding?

**Comment by dejanb** [comment]
@sophokles73 I agree. There's no point in restarting the container if the error will persist. It might be better to indicate this problem in ready/healthy status

**Comment by b-abel** [comment]
Rebased.

File: service-base/src/main/java/org/eclipse/hono/service/limiting/DefaultConnectionLimitManager.java [hunk 1]
```diff
@@ -61,6 +61,10 @@ private int autoconfigureConnectionLimit() {
 
         final int recommendedLimit = strategy.getRecommendedLimit();
 
+        if (recommendedLimit == 0) {
+            throw new RuntimeException("Not enough memory to handle connections. To override this check, configure a connection limit.");
```
Comment (sophokles73) [inline service-base/src/main/java/org/eclipse/hono/service/limiting/DefaultConnectionLimitManager.java]: Please throw a more specific exception and also document it in the constructor's JavaDoc.

Comment (b-abel) [reply service-base/src/main/java/org/eclipse/hono/service/limiting/DefaultConnectionLimitManager.java]: Like this?

Comment (sophokles73) [reply service-base/src/main/java/org/eclipse/hono/service/limiting/DefaultConnectionLimitManager.java]: Sorry, I didn't make myself clear enough. I would have been completely content with e.g. throwing an `IllegalStateException` here. However, using a newly designed exception works as well, of course :-)

**Pull request merged**